### ISSUE TITLE
[Crash] Freezing while updating apps

### ISSUE THREAD

**Issue by vertigo220** [Iamlooker/Droid-ify#244]
**Describe the bug**
App froze and became unresponsive multiple times while updating apps. I hadn't updated apps in a couple weeks, so I sync'd the repos then starting going through updating all the apps. There were at least a dozen updates, and I'd go in one, hit update, go back and go into another, hit update, and so on. At least twice while doing this, Droid-ify stopped responding and I had to force close it. The first time it happened, I got the pop-up from Android saying it had stopped responding and chose to wait, but after a few more minutes it was still frozen, so I tried to get the pop-up again but couldn't, so I had to go to the system page for it and force stop it there. When it happened again (I think on the same app, though I don't remember which app it was and since I was updating several at the same time I can't say for sure, and when I saved that app for last after everything else was updated it didn't freeze), I again couldn't get the system to show the pop-up and had to go to the app details page to force close it. It just seemed like it was getting overwhelmed by queueing too many app updates one right after the other, and I don't think that was the first time this issue has happened.

**To Reproduce**
1. Allow numerous app updates to accumulate
2. Sync the repo (not sure if it matters, but I manually sync'd first, which hadn't been done in at least several days
3. Quickly go through each app and add each one to the update queue without waiting for them to download/update
4. App freezes

**Smartphone:**
 - Device: Samsung A71 5G
 - OS: Android 12
 - Version: 0.5.1 at the time, just updated to 0.5.2

**Comment by Iamlooker** [comment]
Fixed with the new installer which will be available on sunday probably

**Comment by vertigo220** [comment]
Ran into a couple other issues today, hopefully also fixed with the new installer but putting them here just in case:

- Once starting an update, if I switched away from Droid-ify or turned the screen off while it was downloading, when I went back to it, it would be stuck at the download progress it was at when I left it and wouldn't change, and I had to go back to the main screen (updates tab) then back to the app page to get it to refresh, at which point it would either be done or I'd have to click the update button again and it would install (meaning it was done downloading even though it appeared to only be partially done). This happened multiple times and was very consistent.
- Once I was finished doing all the updates, the notification was stuck showing the animated arrow for hours due to showing that a couple apps were ready to install, despite the fact I'd already completed their updates/installs. Only once I tapped their notifications did the icon go away. This is the same issue reported in #199 (though not sure if the same cause) which was supposedly fixed in the latest alpha version, and at the time I was on 0.5.1 and am now on 0.5.5, so it's unclear if that fix simply still hasn't made it to release or if it did and the problem is still there despite it.

**Comment by vertigo220** [comment]
Updated to 0.5.6 today and already had it crash once and freeze once, both while updating. Also, while the first issue in my last post doesn't seem to be the same anymore, it's not completely fixed. Now it tends to remove the download progress bar instead of leaving it showing stuck where it was (though it still did this once out of several times), but the button is stuck showing "Cancel" and when I go back to the list the app is gone, i.e. it's updated. So what appears to be happening is the same as I mentioned in my previous post, only Droid-ify is now better at completing the update/install, so most of them are completing instead of needing me to manually do it like before, but the button isn't updating to say "Launch" once they're updated, and when they're unable to complete on their own, the progress bar still shows and I have to go back then reselect the app from the list to go back to its page and click the update button to get it to install.

**Comment by vertigo220** [comment]
Just crashed again. Hit install on an app, it downloaded it and started to install and just closed. I reopened it and the app wasn't installed, so I hit install again and it worked.

@Iamlooker I assumed you'd see these updates, but you haven't replied or reopened the issue so mentioning you to make sure you see them.

**Comment by vertigo220** [comment]
Also, not sure if this is related (similar to the first point in my second post) or if it should be its own issue, but when I initiated the install of a couple apps while another one was downloading for a Play Store install, they said they were waiting, which sort of makes sense, since the Play Store was "installing" an app (though it seems that Droid-ify shouldn't be held up if it's not actively installing but just downloading, though I can see how that could get tricky). The problem was that after the other app finished, I checked Droid-ify expecting the apps to have automatically commenced, but they still said waiting, and I couldn't get them to start. I had to close and reopen Droid-ify to get them to download and install.
